testhelper/e2e_tests.go

```diff
@@ -24,13 +24,47 @@ func MakeSubtestName(testId, subtestId, testName string) string {
 }
 
 func SetupEndToEndWithInstall(t *testing.T, pkgrConfig, testLibrary string) {
+	SetupEndToEndWithInstallFull(t, pkgrConfig, testLibrary, nil, "", false)
+}
+
+// SetupEndToEndWithInstallFull is like SetupEndToEndWithInstall but
+// takes three additional arguments---env, dir, and expectFail---and
+// returns the combined output.
+//
+// env and dir are set as the environment and working directory for
+// the `pkgr install` call.  When expectFail is true, t.Fatalf() is
+// called if the command does _not_ fail.
+func SetupEndToEndWithInstallFull(
+	t *testing.T,
+	pkgrConfig,
+	testLibrary string,
+	env []string,
+	dir string,
+	expectFail bool,
+) (string, error) {
 	DeleteTestFolder(t, testLibrary)
 
 	installCmd := command.New("pkgr", "install", fmt.Sprintf("--config=%s", pkgrConfig))
-	err := installCmd.Run()
-	if err != nil {
-		t.Fatalf("could not install baseline packages to '%s' with config file '%s'. Error: %s", testLibrary, pkgrConfig, err)
+	if len(env) > 0 {
+		installCmd.Env = env
 	}
+	if len(dir) > 0 {
+		installCmd.Dir = dir
+	}
+
+	out, err := installCmd.CombinedOutput()
+
+	if expectFail {
```

Comment:
This logic, while correct, was a bit confusing to me.

I would propose one of the following revisions for clarity.

```
if err != nil && !expectError {
  t.Fatalf("could not install baseline packages to '%s' with config file '%s'. Error: %s", testLibrary, pkgrConfig, err)
}

if err == nil && expectError {
  t.Fatalf("install to '%s' with config file '%s' unexpectedly succeeded", testLibrary, pkgrConfig)
}
```
or
```
if err != nil {
  if !expectFail {
    t.Fatalf("could not install baseline packages to '%s' with config file '%s'. Error: %s", testLibrary, pkgrConfig, err)
  }
} else {
  if expectFail {
    t.Fatalf("install to '%s' with config file '%s' unexpectedly succeeded", testLibrary, pkgrConfig)
  }
}
```